build_byod_snapshot.sh

```diff
@@ -0,0 +1,319 @@
+#!/bin/bash
+
+set -uo pipefail
+
+# planned improvement will be commented in the following task
+# https://anyscale1.atlassian.net/browse/AE-529
+
+print_level() {
+    t=$(date '+%Y-%m-%d %H:%M:%S')
+    log_level=$1
+    shift
+    echo "[$t][$log_level] -- $@"
+}
+
+print_info() {
+    print_level info $@
+}
+
+pause() {
+    while true; do
+        read -rp "Input Y to continue, N to exit, and S to skip: " input
+        case "$input" in
+            [Yy]* ) return 1;;  # Continue
+            [Nn]* ) echo "Exiting..."; exit 0;;  # Exit the script
+            [Ss]* ) return 2;;  # Skip
+            * ) echo "Invalid input, please try again.";;
+        esac
+    done
+}
+
+assume_role() {
+    read -r ak sa st <<< $(
+        aws sts assume-role \
+        --role-arn $1 \
+        --role-session-name ${USER}DockerImagePush \
+        --query 'Credentials.[AccessKeyId,SecretAccessKey,SessionToken]' \
+        --output text
+    )
+    export AWS_ACCESS_KEY_ID=$ak
+    export AWS_SECRET_ACCESS_KEY=$sa
+    export AWS_SESSION_TOKEN=$st
+}
+
+assume_bootstrap_role() {
+    # to assume this role one should have credentials from anyscale-dev-product account
+    assume_role arn:aws:iam::830883877497:role/bk-premerge-first-jawfish-large-Role
+}
+
+check_role() {
+    aws sts get-caller-identity
+}
+
+setup_local_registry() {
+    # Run the local docker registry
+    print_info "Starting local Docker registry"
+    docker run -d -p $DOCKER_REGISTRY_PORT:5000 --restart always --name registry registry:2
+    print_info "Local Docker registry started, listening port $DOCKER_REGISTRY_PORT"
+}
+
+setup_tool_repo() {
+    pushd $TMP_REPO_DIR
+    print_info "Cloning ray-project/ray.git"
+    # TODO @timy remove this once dockerfile is placed in anyscale/templates repo
+    git clone git@github.com:ray-project/ray.git
+    print_info "Cloning anyscale/product.git"
+    git clone git@github.com:anyscale/product.git
+    popd # TMP_REPO_DIR
+}
+
+setup_crane() {
+    print_info "Downloading crane from ${CRANE_URL}"
+    curl -sL "${CRANE_URL}" | tar xzf - -C /tmp/
```

Comment:
```suggestion
    curl -sSfL "${CRANE_URL}" | tar xzf - -C /tmp/
```
Nit: to report error status codes to pipefail via rc=1